contracts/contracts/UmbraPaymaster.sol

```diff
@@ -0,0 +1,41 @@
+pragma solidity ^0.6.2;
+pragma experimental ABIEncoderV2;
+
+import "@opengsn/gsn/contracts/BasePaymaster.sol";
+
+contract UmbraPaymaster is BasePaymaster {
+
+    address public umbraAddr;
+
+    constructor(address _umbraAddr) public {
+        umbraAddr = _umbraAddr;
+    }
+
+    function acceptRelayedCall(
+		GSNTypes.RelayRequest calldata relayRequest  ,
+		bytes calldata approvalData,
+		uint256 maxPossibleGas
+	) external view override returns (bytes memory context) {
+		(approvalData, maxPossibleGas);  // avoid a warning
+
+		require(relayRequest.target == umbraAddr, "UmbraPaymaster: Not Target");
+
+		return abi.encode(0x0);
```

Comment:
Right now this will relay all calls. I think we want to only relay calls to the `withdrawToken` function. Here's an example of how to do this from Dror in the hack money discord

```solidity
require( reqlayRequest.target == address(myTarget));
bytes4 sig = GsnUtils.getMethodSig(relayRequest.encodedFunction); 
require(sig == myTarget.withdarw.selector || sig == myTarget.deposit.selector);
```

Though, we could always add this later and create an issue to track it

Reply:
Good call, created #20 to tackle this. This is also where we'd implement the Uniswap swap to ETH to pay for the gas.